### ISSUE TITLE
[Bug]: Ghostscript rasterizing failed

### ISSUE THREAD

**Issue by JoKalliauer** [ocrmypdf/OCRmyPDF#1351]
### Describe the bug

I think it is a Ghostscript-bug, however it has been reported here several times #1331 #1324 #952 [Actually different issues -j]

```bash
    8 Weighted average image DPI is 0.9, max DPI is 15.3. The discrepancy may indicate a    _pipeline.py:477
high detail region on this page, but could also indicate a problem with the input PDF file.
Page image will be rendered at 0.9 DPI.
    8 Unrecoverable error: rangecheck in setscreenOperand stack:                          ghostscript.py:139
    0.0544299  0  --nostringval--

OCR                   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━╺━━━━  88% 7/8 0:00:01
SubprocessOutputError: Ghostscript rasterizing failed  
```

### Steps to reproduce

```plain text
wget https://github.com/user-attachments/files/16145703/broken.pdf
qpdf --check broken.pdf
ocrmypdf broken.pdf output.pdf --skip-text
```

### Files

[broken.pdf](https://github.com/user-attachments/files/16145703/broken.pdf)

### How did you download and install the software?

Linux package manager (apt, dnf, etc.)

### OCRmyPDF version

16.4.1

### Relevant log output

```plain text
ocrmypdf 16.4.1                                                                               __main__.py:59
Running: ['tesseract', '--version']                                                          __init__.py:133
Found tesseract 5.4.1                                                                        __init__.py:343
Running: ['tesseract', '--version']                                                          __init__.py:133
Running: ['tesseract', '--version']                                                          __init__.py:133
Running: ['gs', '--version']                                                                 __init__.py:133
Found gs 10.2.1                                                                              __init__.py:343
Running: ['gs', '--version']                                                                 __init__.py:133
Running: ['tesseract', '--list-langs']                                                       __init__.py:133
stdout/stderr = List of available languages in "/usr/share/tesseract-ocr/5/tessdata/" (3):    __init__.py:73
deu
eng
osd

pikepdf mmap enabled                                                                          helpers.py:326
os.symlink(broken.pdf, /tmp/ocrmypdf.io.ymm5zzvi/origin)                                      helpers.py:179
os.symlink(/tmp/ocrmypdf.io.ymm5zzvi/origin, /tmp/ocrmypdf.io.ymm5zzvi/origin.pdf)            helpers.py:179
Gathering info with 1 thread workers                                                             info.py:778
pikepdf mmap enabled                                                                          helpers.py:326
Scanning contents     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 100% 8/8 0:00:00
Using Tesseract OpenMP thread limit 1                                                   tesseract_ocr.py:199
Start processing 8 pages concurrently                                                              ocr.py:96
pikepdf mmap enabled                                                                          helpers.py:326
    1 skipping all processing on this page                                                  _pipeline.py:330
    2 skipping all processing on this page                                                  _pipeline.py:330
pikepdf mmap enabled                                                                          helpers.py:326
pikepdf mmap enabled                                                                          helpers.py:326
    3 skipping all processing on this page                                                  _pipeline.py:330
    4 skipping all processing on this page                                                  _pipeline.py:330
    5 skipping all processing on this page                                                  _pipeline.py:330
pikepdf mmap enabled                                                                          helpers.py:326
    6 skipping all processing on this page                                                  _pipeline.py:330
pikepdf mmap enabled                                                                          helpers.py:326
    7 skipping all processing on this page                                                  _pipeline.py:330
    3 Text rotation: (text, autorotate, content) -> text misalignment = (0, 0, 0) -> 0         _graft.py:140
    8 Rasterize with pngmono, rotation 0                                                    _pipeline.py:539
    3 Page rotation: (content, auto) -> page = (0, 0) -> 0                                     _graft.py:165
    8 Weighted average image DPI is 0.9, max DPI is 15.3. The discrepancy may indicate a    _pipeline.py:477
high detail region on this page, but could also indicate a problem with the input PDF file.
Page image will be rendered at 0.9 DPI.
    1 Text rotation: (text, autorotate, content) -> text misalignment = (0, 0, 0) -> 0         _graft.py:140
    8 Running: ['gs', '-dQUIET', '-dSAFER', '-dBATCH', '-dNOPAUSE',                          __init__.py:133
'-dInterpolateControl=-1', '-sDEVICE=pngmono', '-dFirstPage=8', '-dLastPage=8',
'-r0.870933x0.870933', '-dPDFSTOPONERROR', '-o', '-', '-sstdout=%stderr',
'-dAutoRotatePages=/None', '-f', '/tmp/ocrmypdf.io.ymm5zzvi/origin.pdf']
    1 Page rotation: (content, auto) -> page = (0, 0) -> 0                                     _graft.py:165
    2 Text rotation: (text, autorotate, content) -> text misalignment = (0, 0, 0) -> 0         _graft.py:140
    2 Page rotation: (content, auto) -> page = (0, 0) -> 0                                     _graft.py:165
    4 Text rotation: (text, autorotate, content) -> text misalignment = (0, 0, 0) -> 0         _graft.py:140
    4 Page rotation: (content, auto) -> page = (0, 0) -> 0                                     _graft.py:165
    5 Text rotation: (text, autorotate, content) -> text misalignment = (0, 0, 0) -> 0         _graft.py:140
    5 Page rotation: (content, auto) -> page = (0, 0) -> 0                                     _graft.py:165
    6 Text rotation: (text, autorotate, content) -> text misalignment = (0, 0, 0) -> 0         _graft.py:140
    6 Page rotation: (content, auto) -> page = (0, 0) -> 0                                     _graft.py:165
    7 Text rotation: (text, autorotate, content) -> text misalignment = (0, 0, 0) -> 0         _graft.py:140
    7 Page rotation: (content, auto) -> page = (0, 0) -> 0                                     _graft.py:165
    8 stderr = Unrecoverable error: rangecheck in setscreenOperand stack:                     __init__.py:75
    0.0544299  0  --nostringval--

    8 Unrecoverable error: rangecheck in setscreenOperand stack:                          ghostscript.py:139
    0.0544299  0  --nostringval--

OCR                   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━╺━━━━  88% 7/8 0:00:01
ExitCodeException                                                                             _common.py:259
Traceback (most recent call last):
  File "/home/jkalliau/.local/lib/python3.10/site-packages/ocrmypdf/_exec/ghostscript.py",
line 137, in rasterize_pdf
    p = run(args_gs, stdout=PIPE, stderr=PIPE, check=True)
  File "/home/jkalliau/.local/lib/python3.10/site-packages/ocrmypdf/subprocess/__init__.py",
line 62, in run
    proc = subprocess_run(args, env=env, check=check, **kwargs)
  File "/usr/lib/python3.10/subprocess.py", line 526, in run
    raise CalledProcessError(retcode, process.args,
subprocess.CalledProcessError: Command '['gs', '-dQUIET', '-dSAFER', '-dBATCH', '-dNOPAUSE',
'-dInterpolateControl=-1', '-sDEVICE=pngmono', '-dFirstPage=8', '-dLastPage=8',
'-r0.870933x0.870933', '-dPDFSTOPONERROR', '-o', '-', '-sstdout=%stderr',
'-dAutoRotatePages=/None', '-f', '/tmp/ocrmypdf.io.ymm5zzvi/origin.pdf']' returned non-zero
exit status 255.

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/home/jkalliau/.local/lib/python3.10/site-packages/ocrmypdf/_pipelines/_common.py",
line 249, in cli_exception_handler
    return fn(options, plugin_manager)
  File "/home/jkalliau/.local/lib/python3.10/site-packages/ocrmypdf/_pipelines/ocr.py", line
190, in _run_pipeline
    optimize_messages = exec_concurrent(context, executor)
  File "/home/jkalliau/.local/lib/python3.10/site-packages/ocrmypdf/_pipelines/ocr.py", line
117, in exec_concurrent
    executor(
  File "/home/jkalliau/.local/lib/python3.10/site-packages/ocrmypdf/_concurrent.py", line 78,
in __call__
    self._execute(
  File
"/home/jkalliau/.local/lib/python3.10/site-packages/ocrmypdf/builtin_plugins/concurrency.py",
line 144, in _execute
    result = future.result()
  File "/usr/lib/python3.10/concurrent/futures/_base.py", line 451, in result
    return self.__get_result()
  File "/usr/lib/python3.10/concurrent/futures/_base.py", line 403, in __get_result
    raise self._exception
  File "/usr/lib/python3.10/concurrent/futures/thread.py", line 58, in run
    result = self.fn(*self.args, **self.kwargs)
  File "/home/jkalliau/.local/lib/python3.10/site-packages/ocrmypdf/_pipelines/ocr.py", line
78, in _exec_page_sync
    ocr_image_out, pdf_page_from_image_out, orientation_correction = process_page(
  File "/home/jkalliau/.local/lib/python3.10/site-packages/ocrmypdf/_pipelines/_common.py",
line 391, in process_page
    ocr_image, preprocess_out = make_intermediate_images(
  File "/home/jkalliau/.local/lib/python3.10/site-packages/ocrmypdf/_pipelines/_common.py",
line 327, in make_intermediate_images
    rasterize_out = rasterize(
  File "/home/jkalliau/.local/lib/python3.10/site-packages/ocrmypdf/_pipeline.py", line 543,
in rasterize
    page_context.plugin_manager.hook.rasterize_pdf_page(
  File "/home/jkalliau/.local/lib/python3.10/site-packages/pluggy/_hooks.py", line 501, in
__call__
    return self._hookexec(self.name, self._hookimpls.copy(), kwargs, firstresult)
  File "/home/jkalliau/.local/lib/python3.10/site-packages/pluggy/_manager.py", line 119, in
_hookexec
    return self._inner_hookexec(hook_name, methods, kwargs, firstresult)
  File "/home/jkalliau/.local/lib/python3.10/site-packages/pluggy/_callers.py", line 138, in
_multicall
    raise exception.with_traceback(exception.__traceback__)
  File "/home/jkalliau/.local/lib/python3.10/site-packages/pluggy/_callers.py", line 102, in
_multicall
    res = hook_impl.function(*args)
  File
"/home/jkalliau/.local/lib/python3.10/site-packages/ocrmypdf/builtin_plugins/ghostscript.py",
line 105, in rasterize_pdf_page
    ghostscript.rasterize_pdf(
  File "/home/jkalliau/.local/lib/python3.10/site-packages/ocrmypdf/_exec/ghostscript.py",
line 140, in rasterize_pdf
    raise SubprocessOutputError('Ghostscript rasterizing failed') from e
ocrmypdf.exceptions.SubprocessOutputError: Ghostscript rasterizing failed
```

**Comment by jbarlow83** [comment]
Please note this issue is not related to the other linked issues, apart from having one line in the error log in common. One was a corrupt PDF, and the other two were probably also corrupt PDFs but the submitters did not provide a test file.

This one is fixable since ocrmypdf miscalculates the rendering resolution and then Ghostscript refuses to work with that value.

**Comment by JoKalliauer** [comment]
Confirm fix in development-Version `16.4.2.dev7+g621d6a0`
